src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/ReuploadFailedEnvelopeTask.java

```diff
@@ -55,29 +55,33 @@ public FailedDocUploadProcessor getProcessor() {
     @SchedulerLock(name = "re-upload-failures")
     @Scheduled(fixedDelayString = "${scheduling.task.reupload.delay}")
     public void processUploadFailures() throws InterruptedException {
-        ExecutorService executorService = Executors.newFixedThreadPool(
-            accessMapping.size(),
-            r -> new Thread(r, "BSP-REUPLOAD-%d")
-        );
-        CompletionService<Void> completionService = new ExecutorCompletionService<>(executorService);
-
-        accessMapping
-            .parallelStream()
-            .map(Mapping::getJurisdiction)
-            .forEach(jurisdiction -> {
-                FailedDocUploadProcessor processor = getProcessor();
-
-                completionService.submit(() -> {
-                    log.info("Processing failed documents for jurisdiction {}", jurisdiction);
-
-                    processor.processJurisdiction(jurisdiction);
-
-                    return null;
+        log.info("Started failed document processing job");
+
+        try {
+            ExecutorService executorService = Executors.newFixedThreadPool(
+                accessMapping.size(),
+                r -> new Thread(r, "BSP-REUPLOAD-%d")
```

Comment:
perhaps same thing - no need of a suffix here?

Reply:
would be nice, yes

Reply:
Made the change here: https://github.com/hmcts/bulk-scan-processor/pull/384 . The suffix is thread number (within the pool), starting from 0.